functions/v2/typed/greeting/main.py

```diff
@@ -0,0 +1,60 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# [START functions_typed_greeting]
+import functions_framework
+
+
+class GreetingRequest:
+    first_name: str
+    last_name: str
+
+    # Required to deserialize the request
+    @staticmethod
+    def from_dict(req: dict) -> "GreetingRequest":
+        obj = GreetingRequest()
+        obj.first_name = req["first_name"]
+        obj.last_name = req["last_name"]
+        return obj
+
+
+class GreetingResponse:
+    message: str
+
+    def __init__(self, message: str):
+        self.message = message
+
+    # Required to serialize the response
+    def to_dict(self) -> dict:
+        return {
+            "message": self.message,
+        }
+
+    # Used for testing
+    def __eq__(self, other):
+        return self.message == other.message
+
+
+@functions_framework.typed
+def greeting(req: GreetingRequest) -> GreetingResponse:
+    return GreetingResponse(
+        "Hello {first_name} {last_name}!".format(
+            first_name=req.first_name,
+            last_name=req.last_name,
+        )
```

Comment:
unless used in logging, f-strings are preferred. See https://google.github.io/styleguide/pyguide.html#310-strings
```suggestion
        f"Hello {req.first_name} {req.last_name}!"
```